ignite/handlers/checkpoint.py

```diff
@@ -801,52 +816,40 @@ def _check_and_setup(dirname: Path, create_dir: bool, require_empty: bool) -> No
                 )
 
     def __call__(self, checkpoint: Mapping, filename: str, metadata: Optional[Mapping] = None) -> None:
-        path = self.dirname / filename
-
-        if idist.has_xla_support:
-            self._save_xla(checkpoint, path)
-        else:
-            self._save_native(checkpoint, path)
+        if self.save_on_rank == idist.get_rank():
+            path = self.dirname / filename
 
-    @idist.one_rank_only()
-    def _save_native(self, checkpoint: Mapping, path: Path) -> None:
-        self._save_func(checkpoint, path, torch.save)
+            if idist.has_xla_support:
+                import orch_xla.core.xla_model as xm
 
-    def _save_xla(self, checkpoint: Mapping, path: Path) -> None:
-        import torch_xla.core.xla_model as xm
-
-        # all tpu procs should enter here as internally performs sync across device
-        self._save_func(checkpoint, path, xm.save, rank=idist.get_rank())
+                # all tpu procs should enter here as internally performs sync across device
+                self._save_func(checkpoint, path, xm.save)
```

Comment:
For XLA all procs should enter xm.save, but now you added `if self.save_on_rank == idist.get_rank():` on the top, so only one proc will enter this function. Am I missing something?

Reply:
I corrected it